src/sources/stdin.rs

```diff
@@ -94,38 +99,38 @@ where
         let mut out =
             out.sink_map_err(|error| error!(message = "Unable to send event to out.", %error));
 
-        let res = receiver
+        let mut lines = receiver
             .take_until(shutdown)
-            .map_err(|error| emit!(StdinReadFailed { error }))
-            .map_ok(move |line| {
-                emit!(StdinEventReceived {
-                    byte_size: line.len()
-                });
-                create_event(Bytes::from(line), &host_key, &hostname)
-            })
-            .forward(&mut out)
-            .inspect(|_| info!("Finished sending."))
-            .await;
+            .map_err(|error| emit!(StdinReadFailed { error }));
 
-        let _ = out.flush().await; // error emitted by sink_map_err
+        while let Some(Ok(line)) = lines.next().await {
+            emit!(StdinEventReceived {
+                byte_size: line.len()
+            });
```

Comment:
Do we want to do something with the error if we get `Some(Err(_))` here? If not, it might be worth noting why not.

Reply:
Our `Decoder` already emits logs if framing/parsing failed:

https://github.com/timberio/vector/blob/f6c8b31fcd95d1ec1e400063c74672e24b2fc3e0/src/codec/mod.rs#L74

https://github.com/timberio/vector/blob/f6c8b31fcd95d1ec1e400063c74672e24b2fc3e0/src/codec/mod.rs#L90

It forwards the error in case the components needs to handle it (e.g. returning an error code in `HTTP`).

Will add a comment on why it's not needed here

Reply:
Added a comment:

https://github.com/timberio/vector/blob/b0562481ab153b31151320f4afb499d8c757c790/src/sources/stdin.rs#L141-L142